### PR TITLE
Feature/cleanup j client api

### PR DESCRIPTION
**Pull request by dothebart**
# Pull Request Guidelines

Pull requests are an essential collaborative tool for modern software development. 

The below list is intended to help you figure out whether your code is ready to be reviewed
and merged into ArangoDB. The overarching goal is to:

- Reduce the amount of *recurring* defects
- Reduce the impact to the other developer’s time and energy spent on defects in your code
- Increase the overall autonomy and productivity of individual developers

### Scope & Purpose
- [x] The behavior in this PR can be (and was) *manually tested* (support / qa / customers can test it)
- [x] The behaviour change can only be verified via automatic tests

### Testing & Verification
This change is a trivial rework / code cleanup without any test coverage.
This change is already covered by existing tests, such as `tests/js/common/shell/shell-keygen.js`, `tests/js/client/permissions/harden.js`

### CODE REVIEW

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/5178/

File: js/client/bootstrap/modules/internal.js [hunk 1]
```diff
@@ -98,7 +98,51 @@ let appendHeaders = function(appender, headers) {
...
+  // //////////////////////////////////////////////////////////////////////////////
+
+  exports.isCluster = function () {
+    if (exports.arango) {
+      const arangosh = require('@arangodb/arangosh');
+      let requestResult = exports.arango.GET("/_admin/server/role");
+      arangosh.checkRequestResult(requestResult);
+      return requestResult.role === "COORDINATOR";
+    }
+    throw 'not connected';
```
Comment (ObiWahn) [inline js/client/bootstrap/modules/internal.js]: `ArangoError(requestResult)`?

File: js/client/bootstrap/modules/internal.js [hunk 2]
```diff
@@ -98,7 +98,51 @@ let appendHeaders = function(appender, headers) {
...
+  }
+
+  exports.processStatistics = function () {
+    if (exports.arango) {
+      const arangosh = require('@arangodb/arangosh');
+      let requestResult = exports.arango.GET('/_admin/statistics');
+      arangosh.checkRequestResult(requestResult);
+      return requestResult.system;
+    }
+    throw 'not connected';
```
Comment (ObiWahn) [inline js/client/bootstrap/modules/internal.js]: ArangoError?

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/5181/

File: js/client/bootstrap/modules/internal.js (hunk 2 above)
Comment (dothebart) [reply js/client/bootstrap/modules/internal.js]: these throws never triggered - removing them.

File: js/client/bootstrap/modules/internal.js [hunk 3]
```diff
@@ -68,93 +68,102 @@ let appendHeaders = function(appender, headers) {
...
-        var wfc = waitForCollector ? 'true' : 'false';
-        exports.arango.PUT('/_admin/wal/flush?waitForSync=' + wfs + '&waitForCollector=' + wfc, null);
-        return;
-      }
-
-      throw 'not connected';
+      var wfs = waitForSync ? 'true' : 'false';
+      var wfc = waitForCollector ? 'true' : 'false';
+      exports.arango.PUT('/_admin/wal/flush?waitForSync=' + wfs + '&waitForCollector=' + wfc, null);
+      return;
```
Comment (jsteemann) [inline js/client/bootstrap/modules/internal.js]: the `return` statement is now unnecessary.
```suggestion
```

File: js/client/bootstrap/modules/internal.js [hunk 4]
```diff
@@ -68,93 +68,102 @@ let appendHeaders = function(appender, headers) {
...
 
   exports.reloadAqlFunctions = function () {
-    if (exports.arango) {
-      exports.arango.POST('/_admin/aql/reload', null);
-      return;
-    }
-
-    throw 'not connected';
+    exports.arango.POST('/_admin/aql/reload', null);
+    return;
```
Comment (jsteemann) [inline js/client/bootstrap/modules/internal.js]: the `return` statement is now unnecessary.
```suggestion
```

File: js/client/bootstrap/modules/internal.js [hunk 5]
```diff
@@ -68,93 +68,102 @@ let appendHeaders = function(appender, headers) {
...
 
   exports.reloadRouting = function () {
-    if (exports.arango) {
-      exports.arango.POST('/_admin/routing/reload', null);
-      return;
-    }
-
-    throw 'not connected';
+    exports.arango.POST('/_admin/routing/reload', null);
+    return;
```
Comment (jsteemann) [inline js/client/bootstrap/modules/internal.js]: the `return` statement is now unnecessary.
```suggestion
```

**Comment by jsteemann** [comment]
https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/5299/

**Pull request merged**